components/edit_post_modal/edit_post_modal.jsx

```diff
@@ -98,7 +98,7 @@ export default class EditPostModal extends React.PureComponent {
     }
 
     getContainer = () => {
-        return this.refs.editModalBody;
+        return this.refs.emojiPickerContainer;
```

Comment:
why replacing this? what will happen with the editModalBody, is anything else using it? I can't seem to follow the reason.

Perhaps best to create another function for this? even better define the ref like `ref={this.getEmojiPickerRef}` and then

```js
getEmojiPickerRef = (ref) => {
    this.emojiPickerContainer = ref;
}
```

and you'll be able to pass `this.emojiPickerContainer` as the prop below

Reply:
Good question!

The `getContainer()` function appears to only be in use by the `EmojiPickerOverlay` as does the `editModalBody` ref, so I had decided to edit it directly instead of creating a new function. This technique is also currently used by the existing `target` property of the picker. Let me know if you would still like to update! Unfortunately I don't quite follow where you are going with the above (apologies), so I might need some followup if you would still like to go that route.

My thought process for the rest of this approach:

Of all the places I could find where the picker is used, the edit modal is the only place that the picker still needs to be contained rather than being allowed to render at the top of the DOM (like modals, tool tips etc). In this context the picker should probably remain anchored to the edit modal on desktop screen sizes as the modal can move around horizontally when the window is sized and should take the picker along for the ride.

I created the new `emojiPickerContainer` div, as the original container (`editModalBody`) prevented the picker from being able to be sized to full screen on mobile as is. I initially considered using the parent `Modal` component as an alternate container for the picker, but it is being used to horizontally centre a generated div that contains the actual modal contents. The new container can fill the edit modal with minimal impact to existing markup, styling and functionality, allows the picker to go fullscreen on mobile and keeps the picker attached to the modal on desktop.

An alternative to the new container might be to add a class to the `Modal` component when the picker is visible that temporarily changes `position: relative` to `position: static`, allowing the picker to be sized to the overall modal container on mobile screens instead of just the modal body. That would leave most of the markup and functionality of the edit modal as is, leaving the heavy lifting to CSS.

Let me know what you'd like to do when you have a minute! Happy to jump on a call/chat to discuss if needed.

Thx!

Reply:
sounds good, and makes total sense.

Thanks for your response